src/main/java/com/google/javascript/cl2dts/DeclarationGenerator.java

```diff
@@ -513,10 +513,28 @@ public Void caseTemplatizedType(TemplatizedType type) {
             emit("[]");
             return null;
           }
+          Iterator<JSType> it = type.getTemplateTypes().iterator();
+          if ("Object".equals(referencedType.getReferenceName())) {
+            emit("{ [");
+            // TS allows only number or string as index type of an object
+            // https://github.com/Microsoft/TypeScript/blob/master/doc/spec.md#3.9.4
+            if (it.next().isNumberValueType()) {
+              emit("n: number");
+            } else {
+              emit("s: string");
+            }
```

Comment:
should we warn if the input was anything outside `string|number`, so users are not surprised?